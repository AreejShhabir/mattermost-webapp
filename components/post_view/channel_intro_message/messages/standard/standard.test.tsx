// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {channel} from '../test_utils';

import StandardIntroMessage from './standard';

const component = (otherProps: any) => {
    return (
        <StandardIntroMessage
            channel={channel}
            locale={'en'}
            creatorName={'creatorName'}
            usersLimit={10}
            {...otherProps}
        />
    );
};

describe('components/post_view/ChannelIntroMessages', () => {
    describe('test Open Channel', () => {
        test('should match snapshot', () => {
            expect(shallow(component({ }))).toMatchSnapshot();
        });
    });
});
