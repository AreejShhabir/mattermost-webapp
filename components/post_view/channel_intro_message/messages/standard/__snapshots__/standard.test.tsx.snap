// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/ChannelIntroMessages test Open Channel should match snapshot, with boards 1`] = `
<Fragment>
  <h2
    className="channel-intro__title"
  >
    <MemoizedFormattedMessage
      defaultMessage="Beginning of {name}"
      id="intro_messages.beginning"
      values={
        Object {
          "name": "test channel",
        }
      }
    />
  </h2>
  <p
    className="channel-intro__content"
  >
    <MemoizedFormattedMessage
      defaultMessage="This is the start of the {name} channel, created by {creator} on {date}."
      id="intro_messages.creator"
      values={
        Object {
          "creator": "creatorName",
          "date": <FormattedDate
            day="2-digit"
            month="long"
            value={1508265709607}
            year="numeric"
          />,
          "name": "test channel",
        }
      }
    />
    <MemoizedFormattedMessage
      defaultMessage=" Any member can join and read this channel."
      id="intro_messages.anyMember"
    />
    <MemoizedFormattedMessage
      defaultMessage=" This channel's purpose is: {purpose}"
      id="intro_messages.purpose"
      values={
        Object {
          "purpose": "test",
        }
      }
    />
    <br />
  </p>
  <Memo(AddMembersButton)
    channel={
      Object {
        "create_at": 1508265709607,
        "creator_id": "creator_id",
        "delete_at": 0,
        "display_name": "test channel",
        "header": "test",
        "id": "channel_id",
        "last_post_at": 1508265709635,
        "name": "testing",
        "purpose": "test",
        "team_id": "team-id",
        "type": "O",
        "update_at": 1508265709607,
      }
    }
    createBoard={<Memo(Connect(Component)) />}
    showSetHeader={true}
    usersLimit={10}
  />
</Fragment>
`;

exports[`components/post_view/ChannelIntroMessages test Open Channel should match snapshot, without boards 1`] = `
<Fragment>
  <h2
    className="channel-intro__title"
  >
    <MemoizedFormattedMessage
      defaultMessage="Beginning of {name}"
      id="intro_messages.beginning"
      values={
        Object {
          "name": "test channel",
        }
      }
    />
  </h2>
  <p
    className="channel-intro__content"
  >
    <MemoizedFormattedMessage
      defaultMessage="This is the start of the {name} channel, created by {creator} on {date}."
      id="intro_messages.creator"
      values={
        Object {
          "creator": "creatorName",
          "date": <FormattedDate
            day="2-digit"
            month="long"
            value={1508265709607}
            year="numeric"
          />,
          "name": "test channel",
        }
      }
    />
    <MemoizedFormattedMessage
      defaultMessage=" Any member can join and read this channel."
      id="intro_messages.anyMember"
    />
    <MemoizedFormattedMessage
      defaultMessage=" This channel's purpose is: {purpose}"
      id="intro_messages.purpose"
      values={
        Object {
          "purpose": "test",
        }
      }
    />
    <br />
  </p>
  <Memo(AddMembersButton)
    channel={
      Object {
        "create_at": 1508265709607,
        "creator_id": "creator_id",
        "delete_at": 0,
        "display_name": "test channel",
        "header": "test",
        "id": "channel_id",
        "last_post_at": 1508265709635,
        "name": "testing",
        "purpose": "test",
        "team_id": "team-id",
        "type": "O",
        "update_at": 1508265709607,
      }
    }
    createBoard={<Memo(Connect(Component)) />}
    showSetHeader={true}
    usersLimit={10}
  />
</Fragment>
`;
